torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1745,3 +1745,76 @@ def rref_tensor_is_owner(rref_var):
 
         res = rref_tensor_is_owner(rref_var)
         self.assertEqual(res, False)
+
+    @dist_init
+    def test_rpc_async_op(self):
+        @torch.jit.script
+        def two_args_two_kwargs(
+            first_arg,
+            second_arg,
+            first_kwarg=torch.tensor([3, 3]),
+            second_kwarg=torch.tensor([4, 4]),
+        ):
+            return first_arg + second_arg + first_kwarg + second_kwarg
+
+        if self.rank != 0:
+            return
+
+        @torch.jit.script
+        def rpc_async_in_torchscript(dst_worker_name, args, kwargs):
+            # type: (str, Tuple[Tensor, Tensor], Dict[str, Tensor])
+            fut = rpc.api._invoke_rpc_torchscript(
+                dst_worker_name, two_args_two_kwargs, args, kwargs
+            )
+            ret = fut.wait()
+            return ret
+
+        from torch.testing import FileCheck
+        FileCheck().check("dst_worker_name").run(str(rpc_async_in_torchscript.graph))
+
+        dst_worker_name = "worker{}".format((self.rank + 1) % self.world_size)
+
+        # Case 1. Some kwargs are populated by default values.
+        args = (
+            torch.tensor([1, 1]),
+            torch.tensor([2, 2]),
+        )
+        kwargs = {
+            "first_kwarg": torch.tensor([2, 2]),
+        }
```

Comment:
Can we have test cases where the types of the elements in the args tuple are different and also types for the values in kwargs are different? Ex: args can be (Tensor, str, int), kwargs can be {k1 : Tensor, k2: str, k3:int}

Reply:
JIT does not support multiple types in Dict. The error JIT gives is,

> Dict values must contain only a single type, expected: Tensor but found str instead.

Reply:
Added the test case.

When adding it, I noticed that when preparing the `kwargs` dict, being passed to `_invoke_rpc_torchscript(..)`, must be annotated as `Dict[str, Any]` type.

Otherwise, the error JIT type inference gives is 

> "Dict values must contain only a single type, expected: Tensor but found str instead."